src/dune_rules/compilation_context.ml

```diff
@@ -15,7 +15,7 @@ module Includes = struct
            (let+ libs = requires in
             Command.Args.S
               [ iflags libs Byte
-              ; Hidden_deps (Lib_file_deps.deps libs ~groups:[ Cmi ])
+              ; Hidden_deps (Lib_file_deps.deps libs ~groups:[ Cmi; Cmt; Cmti ])
```

Comment:
This seems wrong to me. if something depends on `x`, it does not mean we need to build `x`'s cmt's and cmti's. Why are we adding the dependency here?

Reply:
I see, the Cmt's are build as soon as a compilation context has the `bin_annot` field set to true which appear to be the case for every CC used in Dune except for Menhir and when building ppx_driver.

So the cmts are already always built by Dune isn't it ? However I should had a check that the current CC has the bin_annot field set to true.

Reply:
Yes, cmt's are essentially always built by dune.

>  However I should add a check that the current CC has the bin_annot field set to true.

I'm not sure why you need this check. Sometimes the cmt's are explicitly turned off for some internally generated modules. That's about the only time when bin_annot is set to false.